spec/requests/v1/pacbio/request_library_spec.rb

```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+
+RSpec.describe 'RequestLibraryController', type: :request do
+
+    context '#update' do
+        let(:tag) { create(:tag) }
+        let(:request_library) { create(:pacbio_request_library) }
+
+        context 'on success' do
+            let(:body) do
+                {
+                data: {
+                    type: "tags",
+                    id: request_library.id,
+                    attributes: {
+                    tag_id: tag.id
+                    }
+                }
+                }.to_json
+            end
+
+            it 'has a ok status' do
+                patch v1_pacbio_request_library_path(request_library), params: body, headers: json_api_headers
+                expect(response).to have_http_status(:ok)
+            end
+
+            it 'returns the correct attributes' do
+                patch v1_pacbio_request_library_path(request_library), params: body, headers: json_api_headers
+                json = ActiveSupport::JSON.decode(response.body)
+                expect(json['data']['id']).to eq request_library.id.to_s
+            end
+        end
+
+        #context 'on failure' cannot happen since there is no validation as you cannot send a requestLibrary without a tag_id
```

Comment:
Still worth adding the check to make sure it is handled in the back end. Server code shouldn't rely on the client codes validation, it should do some itself. In the controlled I think the params `permit` method would fail if it wasnt present so maybe adding a test to check that.

Reply:
Just check the formatting in this file too :) curly braces are on odd lines